test/models/concerns/organization_extensions_test.rb

```diff
@@ -26,5 +26,21 @@ def test_audit_manifest_action
       end
       assert_equal @org.manifest_refreshed_at.to_i, current_time.to_i
     end
+
+    def test_clear_syspurpose_status
+      host = @org.hosts.first
+
+      HostStatus::Status.create(id: 1, host_id: host.id, type: 'Katello::PurposeStatus')
+      HostStatus::Status.create(id: 2, host_id: host.id, type: 'Katello::PurposeAddonsStatus')
+      HostStatus::Status.create(id: 3, host_id: host.id, type: 'Katello::PurposeUsageStatus')
+      HostStatus::Status.create(id: 4, host_id: host.id, type: 'Katello::PurposeRoleStatus')
+      HostStatus::Status.create(id: 5, host_id: host.id, type: 'Katello::PurposeSlaStatus')
+
+      assert_equal HostStatus::Status.count, 5
```

Comment:
For the assertions in this test it's more accurate to say "this host has no purpose statuses left" rather than "there are no status records in the db at all (even though it is true)" because your code change intends to achieve the former, not the latter. If someone came a long and added a fixture for some other host to give it a status, then this test would fail :(

`assert_equal 5, host.host_statuses.count` should be more future-proof and accurate!

Reply:
Updated the test to your suggestion, I see with `puts` how it is more accurate and better to have a test that is adaptive compared to something that could fail down the line.